salt/utils/schema.py

```diff
@@ -515,7 +515,7 @@ def __call__(cls, *args, **kwargs):
                 'Please pass all arguments as named arguments. Un-named '
                 'arguments are not supported'
             )
-        for key in kwargs.keys():
+        for key in kwargs.copy().keys():
```

Comment:
In python2, you can use the .keys() object, and also modify the dictionary that you are using, but in python3, since keys() is a generator, you get an error when you pop off something inside the generator.

```
[root@3f59d2083157 salt]# cat that
from __future__ import print_function
kwargs = {'one': 1,
          'two': 2,
          'three': 3,
          'four': 4,
          'five': 5}

for key in kwargs.keys():
    print(key)
    kwargs.pop(key)
[root@3f59d2083157 salt]# python2 that
four
three
five
two
one
[root@3f59d2083157 salt]# python3 that
three
Traceback (most recent call last):
  File "that", line 8, in <module>
    for key in kwargs.keys():
RuntimeError: dictionary changed size during iteration
```

You also get this error if you use iterkeys() in python2.

```
[root@3f59d2083157 salt]# cat that
from __future__ import print_function
kwargs = {'one': 1,
          'two': 2,
          'three': 3,
          'four': 4,
          'five': 5}

for key in kwargs.iterkeys():
    print(key)
    kwargs.pop(key)
[root@3f59d2083157 salt]# python2 that
four
Traceback (most recent call last):
  File "that", line 8, in <module>
    for key in kwargs.iterkeys():
RuntimeError: dictionary changed size during iteration
```

The copy makes a new top level of the dictionary with references to all the same objects, but you can pop them out of the kwargs object, and the copied one can keep iterating